### PR TITLE
Upgrade CAPI client and removes references to MaxUrlSize

### PR DESCRIPTION
**Pull request by DiegoVazquezNanini**
- Upgrades CAPI client to 10.1 (latest)
- The CAPI client removed MaxUrlSize, therefore, I removed all references to it from the facia client.
- org.json4s.JsonAST.JValue no longer available, so I am changing it to JSValue from play.

/cc @janua @johnduffell 

I need to release this change to finish off a piece of work in MAPI.

Cheers

### CODE REVIEW

File: fapi-client/src/main/scala/com/gu/facia/api/contentapi/IdsSearchQueries.scala [hunk 1]
```diff
@@ -6,14 +6,12 @@ object IdsSearchQueries {
...
   type Url = String
 
-  val MaxUrlSize = com.gu.contentapi.client.Limits.UrlSize
-  //Maximum size that can be requested from CAPI is 50.
   val MaxBatchSize = 50
 
   def makeBatches(ids: Seq[Id])(urlFromIds: Seq[Id] => Url): Option[Seq[Seq[Id]]] = {
     def batchAndRemaining(ids: Seq[Id]): Option[(Seq[Id], Seq[Id])] =
       ids.inits.find{ init =>
-        init.length <= MaxBatchSize && urlFromIds(init).length <= MaxUrlSize
```
Comment (janua) [inline fapi-client/src/main/scala/com/gu/facia/api/contentapi/IdsSearchQueries.scala]: If you are removing `urlFromIds` then you don't need to pass it to `makeBatches`.

File: fapi-client/src/main/scala/com/gu/facia/api/http/HttpResponse.scala [hunk 2]
```diff
@@ -1,11 +1,12 @@
 package com.gu.facia.api.http
 
 import com.gu.facia.api.Response.Async._
-import com.gu.facia.api.json.Json
-import com.gu.facia.api.{HttpError, ApiError, Response}
+import play.api.libs.json.Json
+//import com.gu.facia.api.json.Json
```
Comment (janua) [inline fapi-client/src/main/scala/com/gu/facia/api/http/HttpResponse.scala]: Remove the comment!

File: fapi-client/src/main/scala/com/gu/facia/api/http/HttpResponse.scala [hunk 3]
```diff
@@ -21,11 +22,10 @@ object HttpResponse {
...
   }
 
-  def jsonResponse(request: Request, client: dispatch.Http)(implicit executionContext: ExecutionContext): Response[JValue] = {
+  def jsonResponse(request: Request, client: dispatch.Http)(implicit executionContext: ExecutionContext): Response[JsValue] = {
     for {
       rawResponse <- Right(client(request, HttpResponse.dispatchHandler))
       okResponse <- HttpResponse.okToRight(rawResponse)
-      json <- Json.toJson(okResponse.body)
-    } yield json
+    } yield Json.toJson(okResponse.body)
```
Comment (janua) [inline fapi-client/src/main/scala/com/gu/facia/api/http/HttpResponse.scala]: Does this work the way you expect in the context of `Response`? The other one was throwing exceptions that were being handled by us.

**Comment by DiegoVazquezNanini** [comment]
@janua I think I address all the comments and added the deprecation annotations to the unused HttpResponse. Can you have another look?

**Comment by janua** [comment]
LGTM :+1: 
`HttpResponse` is going next iteration :boom:

**Comment by DiegoVazquezNanini** [comment]
Thanks

**Pull request merged**